lib/ibmdb.js

```diff
@@ -409,32 +425,37 @@ IBMDB.prototype.updateOrCreate = IBMDB.prototype.save =
       // Execution for updateOrCreate requires running two
       // separate SQL statements.  The second depends on the
       // result of the first.
+      var where = {};
+      where[id] = data[id];
```

Comment:
@superkhau and @qpresley
>var where = {
  id: data.id
};

**No do not do that!**

@superkhau I'm afraid what you are proposing breaks lots of things; id/PK is not always called `id`; that's why they need to use the following to add the right name of id/pk in `where` object as the key:
```
where[id] = data[id];
```